images/kernel/Makefile

```diff
@@ -1,11 +1,22 @@
-KERNEL_VERSION ?= 4.19.47
-KERNEL_EXTRA ?= 
+# Check https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/refs/ for updates
+KERNEL_VERSIONS ?= 4.14.166 4.19.97 5.4.13
 
 all: build
-build:
-	docker build -t weaveworks/ignite-kernel:${KERNEL_VERSION}${KERNEL_EXTRA} \
-		--build-arg KERNEL_VERSION=${KERNEL_VERSION} \
-		--build-arg KERNEL_EXTRA=${KERNEL_EXTRA} .
 
-push:
-	docker push weaveworks/ignite-kernel:${KERNEL_VERSION}${KERNEL_EXTRA}
+upgrade: $(addprefix upgrade-,$(KERNEL_VERSIONS))
+	./patch-config.sh
+	for file in generated/*; do \
+		./upgrade-config.sh $$file $$file; done
```

Comment:
Can you help me understand why this is here?
I'm reading that this runs
```bash
     make clean && make mrproper && cp /tmp/.config . && \
     make EXTRAVERSION="" LOCALVERSION= olddefconfig
```
on the already generated config as the input.
That output then overwrites the input file.

Does this cleanup or do something to the generated config that changes it?
Those changes wouldn't end up in the patch configs unless you ran `make upgrade` twice.

Thanks :+1:

Reply:
This basically sanitizes, reorganizes and formats (overwrites) the kernel config for the final generated configs